### PR TITLE
generate and insert public key on deploy

### PR DESCRIPTION
**Pull request by jhoreman**
### Which issue this PR addresses:

Since we now manage SSH access through JIT directly, as opposed to using JIT to collect a permanent key, we can now remove the aforementioned private key. In order to pass ARM validation an authentication mechanism is required however so we're choosing to generate this on demand on deploy. 
Fixes

### What this PR does / why we need it:

Accompanies the SSH Access JIT changes

### Test plan for issue:

Once deployed we can check the key value in authorised keys 

### Is there any documentation that needs to be updated for this PR?

No docs already exist.

### CODE REVIEW

File: pkg/deploy/config.go
Comment (bennerv) [inline pkg/deploy/config.go:122]: one more ask - can you put this in the `validate()` function?  otherwise lgtm

**Review by bennerv — COMMENTED** [review]
one nit

**Review by bennerv — DISMISSED** [review]
LGTM - just need to verify you can SSH to FF Prod with an escort session and get onto the VMSS without using the SSH key.  Once that is done, I'm good to merge.

File: pkg/deploy/config.go
Comment (cadenmarchese) [inline pkg/deploy/config.go:154]: Should we handle the error here?

File: pkg/deploy/config.go
Comment (dem4gus) [inline pkg/deploy/config.go:158]: If I'm understanding the description correctly we no longer need a permanently-stored key for SSH access to the servers, which means we don't care that the generated private key is lost after this block?

File: pkg/deploy/config.go
Comment (facchettos) [reply pkg/deploy/config.go:154]: absolutely. if `err!=nil` you have a nil pointer and will probably panic on the next line.

File: pkg/deploy/config.go
Comment (jhoreman) [reply pkg/deploy/config.go:158]: Correct! We actually want to forget about it and have garbage collection kick in.

File: pkg/deploy/config.go
Comment (jhoreman) [reply pkg/deploy/config.go]: Thanks, both. Added error handling. Please review.

**Review by bennerv — APPROVED** [review]
lgtm - waiting on confirmation vmss JIT works as expected with an escort session then can merge

**Review by ross-bryan — APPROVED** [review]
lgtm

**Review by cadenmarchese — APPROVED** [review]
Thanks!

**Pull request merged**
